src/Aks/Aks/Commands/CreateOrUpdateKubeBase.cs

```diff
@@ -335,7 +335,7 @@ protected void AddAcrRoleAssignment(string acrName, string acrParameterName, Acs
             {
                 try
                 {
-                    var servicePrincipal = GraphClient.ServicePrincipals.GetServicePrincipal(acsServicePrincipal.SpId);
+                    var servicePrincipal = GraphClient.ServicePrincipals.GetServicePrincipal(acsServicePrincipal.ObjectId);
```

Comment:
See line 333-334, `ObjectId` is null here. We have no choice but to use the list API to filter the service principals by application ID